modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/jobs/VersionMigrationsWorker.kt

```diff
@@ -119,51 +126,20 @@ class VersionMigrationsJob : JobService() {
         }
     }
 
-    @Inject lateinit var podcastManager: PodcastManager
-
-    @Inject lateinit var episodeManager: EpisodeManager
-
-    @Inject lateinit var settings: Settings
-
-    @Inject lateinit var fileStorage: FileStorage
-
-    @Inject lateinit var appDatabase: AppDatabase
-
-    @Inject lateinit var playbackManager: PlaybackManager
-
-    @Volatile private var shouldKeepRunning = true
-
-    override fun onStartJob(jobParameters: JobParameters): Boolean {
-        LogBuffer.i(LogBuffer.TAG_BACKGROUND_TASKS, "VersionMigrationsTask onStartJob")
-
-        Thread {
-            var shouldReschedule = false
-            try {
-                if (!shouldKeepRunning) {
-                    shouldReschedule = true
-                } else {
-                    performMigration()
-                }
-            } catch (t: Throwable) {
-                Timber.e(t)
-                LogBuffer.e(LogBuffer.TAG_BACKGROUND_TASKS, t, "VersionMigrationsTask jobFinished failed.")
-            } finally {
-                LogBuffer.i(LogBuffer.TAG_BACKGROUND_TASKS, "VersionMigrationsTask jobFinished shouldReschedule? $shouldReschedule")
-                jobFinished(jobParameters, shouldReschedule)
-            }
-        }.start()
-
-        return true
-    }
-
-    override fun onStopJob(jobParameters: JobParameters): Boolean {
-        LogBuffer.i(LogBuffer.TAG_BACKGROUND_TASKS, "VersionMigrationsTask onStopJob")
-        shouldKeepRunning = false
-
-        return false
+    override suspend fun doWork(): Result = withContext(Dispatchers.IO) {
+        try {
+            LogBuffer.i(LogBuffer.TAG_BACKGROUND_TASKS, "VersionMigrationsWorker - started")
+            performMigrationsAsync()
+            LogBuffer.i(LogBuffer.TAG_BACKGROUND_TASKS, "VersionMigrationsWorker - finished")
+            Result.success()
+        } catch (t: Throwable) {
```

Comment:
It wasn't in the original code but if the migration fails should we revert migrated version code to the previous one?

It's a more complex problem because we don't execute migration one by one like for example SQL migrations so this can get into a bad state anyway if there are multiple migrations for multiple versions and it fails in the middle. So I'm not sure.